lib/errors/DBError.js

```diff
@@ -8,6 +8,8 @@ class DBError extends Error {
     this.name = this.constructor.name;
     this.nativeError = args.nativeError;
     this.client = args.client;
+    this.schema = args.schema;
```

Comment:
Why did you add these here? No other properties like table, constraint, column etc. are listed here.

Reply:
This will cause every DBError to have these fields.

Reply:
That what I understood from this comment: https://github.com/Vincit/db-errors/pull/4#discussion_r268386422.

Should I just drop the fields?

Reply:
The fields should be there when available, just as you implemented, but this particular change (the two assignments) adds the fields a with `undefined` values even when they don't even belong to the error type. Every single error would have

```js
schema: undefined
database: undefined
```
properties. There's a difference between an object not having a field and an object having a filed with undefined value, even though accessing them returns `undefined`  in both cases. For example if you `console.log` those objects, you see a different result.

TLDR;
Just remove these assignments here and move them the the error classes that can have those fields. Like here https://github.com/Vincit/db-errors/blob/master/lib/errors/ForeignKeyViolationError.js

Reply:
Got it thanks, Updated the PR and the typescript definitions